py-rse/ci.Rmd

```diff
@@ -108,168 +94,146 @@ knitr::include_graphics("figures/rse-ci/add-repo.png")
--   Adds a Travis CI build badge to the project's `README.md`.
--   Opens the Travis CI page for the package so that we can activate Travis CI for the repository.
-    (This assumes the package is already on GitHub and that we have a Travis CI account.)
-
-FIXME: add example here of generated YAML file
+Here we've added a setting called `script`
+that tells Travis to run the following list of commands,
+in this case, only `pytest`.
+We can put almost anything here as long as it doesn't need human interaction
+(i.e., doesn't ask questions).
 
-The [usethis](https://usethis.r-lib.org/reference/ci.html) documentation has more details,
-but it really is this simple.
+Getting CI to run the project tests is the main reason to use CI,
+so that any change made can be tested for correctness.
+And because Travis runs in a "clean" environment (aka, not your own computer),
+it can also find problems that don't show up when you test on your own computer.
 
-## How can I use Travis CI to test my project? {#py-rse-ci-testing}
+Before continuing, complete Exercise \@ref(py-rse-ci-ex-add-pytest).
 
-The main reason to set up CI is to test the project every time a change is made.
-We still have to write unit tests (Chapter \@ref(py-rse-correct)),
-but CI makes those tests much more valuable.
+After completing the exercise,
+you'll have seen the Travis CI log that is found below the build overview.
+The log will contain information on what our test command produces,
+whether the test succeeded or failed.
 
-For R packages,
-running tests is built into Travis CI so you don't need to modify anything.
-
-FIXME: er, what? does this mean the commands are auto-generated by `usethis`?
-
-FIXME: Include example here of what failing tests in R look like in Travis CI.
+FIXME: Include example here of what failing tests in Python look like in Travis CI.
 
-For Python packages,
-we need to add `pytest` to the `script` key in the `.travis.yml` file:
+The project will have failed because of a simple reason:
+we use other packages in our Zipf's Law project and Travis doesn't know about them.
+So we need to add an additional line to the `.travis.yml` file:
 
 ```yaml
 language: python
+
 python:
 - "3.6"
+
 install:
 - pip install -r requirements.txt
+
 script:
 - pytest
 ```
 
-With this in place,
-the Travis CI log located right below the build overview
-will contain all of the information that our testing command produces when tests fail.
+The `install` setting tells Travis CI how to install the software we need for our project.
+For Python projects,
+the list of packages the project depends on is listed in a file called `requirements.txt`.
+By using `pip`, these packages are then installed in the Travis environment.
+Complete Exercise \@ref(py-rse-ci-ex-requirements) before continuing.
 
-FIXME: Include example here of what failing tests in Python look like in Travis CI.
+TODO: Will requirements.txt already be created in the Package chapter?
```

Comment:
Who could tell us?

Reply:
To address the todo, it is created there.